src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+			}
+			Initialized = true;
+		}
+
+
+		void Model::ModelLog(std::string fmi_logmessage)
+		{
+			std::string logmessage(fmi_logmessage);
+			cout << logmessage;
+		}
+
+		double Model::Round(double x)
+		{
+			return floor(x + 0.5);
+		}
+
+
+
+
+
+		double Model::Part_press(double P, double W)
+		{
+			// Function to compute partial vapor pressure in [kPa]
+			// From page 6.9 equation 38 in ASHRAE Fundamentals handbook (2005)
+			//   P = ambient pressure [kPa]
+			//   W = humidity ratio [kg/kg dry air]
+
+			return (P * W / (0.62198 + W));
+		}
+
+		double Model::Sat_press(double Tdb)
+		{
+			// Function to compute saturation vapor pressure in [kPa]
+			//ASHRAE Fundamentals handbood (2005) p 6.2, equation 5 and 6
+			//   Tdb = Dry bulb temperature [degC]
+			// Valid from -100C to 200 C
+
+			double  C1 = -5674.5359;
+			double  C2 = 6.3925247;
+			double  C3 = -0.009677843;
+			double  C4 = 0.00000062215701;
+			double  C5 = 2.0747825E-09;
+			double  C6 = -9.484024E-13;
+			double  C7 = 4.1635019;
+			double  C8 = -5800.2206;
+			double  C9 = 1.3914993;
+			double  C10 = -0.048640239;
+			double  C11 = 0.000041764768;
+			double  C12 = -0.000000014452093;
+			double  C13 = 6.5459673;
+			double  Sat_press_val = 0;
+
+			double   TK = Tdb + 273.15;         //Converts from degC to degK
+
+			if (TK <= 273.15)
+			{
+				Sat_press_val = exp(C1 / TK + C2 + C3 * TK + C4 * pow(TK, 2) + C5 * pow(TK, 3) + C6 * pow(TK, 4) + C7 * log(TK)) / 1000;
+			}
+			else
+			{
+				Sat_press_val = exp(C8 / TK + C9 + C10 * TK + C11 * pow(TK, 2) + C12 * pow(TK, 3) + C13 * log(TK)) / 1000;
+			}
+			return Sat_press_val;
+
+		}
+
+
+		double Model::CalcHum_ratio_W(double Tdb, double RH, double P)
+		{
+			// Function to calculate humidity ratio [kg H2O/kg air]
+			// Given dry bulb and wet bulb temperature inputs [degC]
+			// ASHRAE Fundamentals handbood (2005)
+			//   Tdb = Dry bulb temperature [degC]
+			//   RH = Relative Humidity [Fraction or %]
+			//   P = Ambient Pressure [kPa]
+
+			double Pws = Sat_press(Tdb);
+			double Hum_rat = 0.62198 * RH * Pws / (P - RH * Pws);   // Equation 22, 24, p6.8
+			return Hum_rat;
+		}
+
+		double Model::CheckVal_W(double W)
+		{
+			if ((W > 1) || (W < 0))
+			{
+				//issue
+				W = W;
+			}
+			return W;
+		}
+		double Model::CheckVal_T(double T)
+		{
+			if ((T > 50) || (T < 10))
+			{
+				//issue
+				T = T;
+			}
+			return T;
+		}
+
+
+		double Model::CalculateMixedAirTemp()
+		{
+			double Tosa_ref = 37.78; //these can not be hard coded and need to be from config file?
+			double MsaRatio_ref = 0.45;
+			double TdbRA_ref = 25.56;
+			double MixedAirDBTempAtRefCon = TdbRA_ref + MsaRatio_ref*(Tosa_ref - TdbRA_ref);
+			return MixedAirDBTempAtRefCon;
+
+		}
+
+		bool Model::SetStandByMode(CMode* pMode0, CSetting *pStandBy, double Tosa, double Wosa, double Tra, double Wra)
+		{
+			bool error;
+			CModeSolutionSpace* solutionspace = &(pMode0->sol);
```

Comment:
The use of pointers and address operators needs to be justified.  This pushes a new maintenance burden onto the development team members who will be fixing bugs and maintaining this code in the future.